src/dmd/dcast.d

```diff
@@ -3020,130 +3220,112 @@ Lagain:
                     t2 = cd2.type.castMod(t2.mod);
                 }
                 else if (cd1)
+                {
                     t1 = cd1.type;
+                }
                 else if (cd2)
+                {
                     t2 = cd2.type;
+                }
+                else if (checkaliastthis)
+                {
+                    Expression te1 = e1;
+                    Expression te2 = e2;
+                    Type tt1 = t;
+                    int r = mergeAliasThis(sc, &tt1, op, &te1, &te2);
+                    if (r)
+                    {
+                        e1 = te1;
+                        e2 = te2;
+                        t1 = e1.type;
+                        t2 = e2.type;
+                        t = tt1;
+                    }
+                    else
+                    {
+                        goto Lincompatible;
+                    }
```

Comment:
I see the above block of code repeated 3 times in this file.  Perhaps you could isolate this in nested function and then add a documentation comment to describe it.  I think that would help the review.

Reply:
fixed